packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts

```diff
@@ -56,14 +68,21 @@ export function trackViews(location: Location, lifeCycle: LifeCycle) {
 
   function onLocationChange() {
     if (currentView.isDifferentView(location)) {
-      // Renew view on location changes
-      currentView.end()
-      currentView.triggerUpdate()
-      currentView = newView(lifeCycle, location, ViewLoadingType.ROUTE_CHANGE, currentView.url)
-    } else {
-      currentView.updateLocation(location)
-      currentView.triggerUpdate()
+      const { shouldCreateView, viewName } = {
```

Comment:
what about letting customers create view even for cases where we would not?

Reply:
You mean we don't use `isDifferentView` at all and let the user decide?

Reply:
more use `isDifferentView` as a fallback, if the user does not return a value for `shouldCreateView`

Reply:
Makes sense, I will do that.